column2Vec/reaserch/column2Vec_re.py

```diff
@@ -0,0 +1,192 @@
+                                        column2vec_weighted_avg, column2vec_avg, cache)
+
+from column2Vec.reaserch.generate_report import generate_time_report, generate_sim_report, generate_stability_report, \
+    generate_partial_column_report
+from config import configure
+from constants import warning_enable
+from similarity.Comparator import cosine_sim
+
+FUNCTIONS = [column2vec_as_sentence, column2vec_as_sentence_clean, column2vec_as_sentence_clean_uniq,
+             column2vec_avg, column2vec_weighted_avg, column2vec_sum, column2vec_weighted_sum]
+MODEL = 'paraphrase-multilingual-mpnet-base-v2'  # 'bert-base-nli-mean-tokens'
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+model = SentenceTransformer(MODEL)
+
+
+def count_embedding(column1: pd.Series, function, key: str) -> pd.Series:
+    """
+    Count embeddings for column1 using function and key
+    """
+    return function(column1, model, key)
+
+
+def stability_test(column1: pd.Series, function, key: str) -> (bool, str):
+    """
+    Test if function run is stable, it should produce same result every time
+    :param column1: column to test
+    :param function: function to use
+    :param key: key for cache
+    """
+    emb1 = function(column1, model, key)
+    emb2 = function(column1, model, key + "s1")
+    emb3 = function(column1, model, key + "s2")
+    first_second = cosine_sim(emb3, emb2)
+    whole_second = cosine_sim(emb1, emb2)
+    whole_first = cosine_sim(emb3, emb1)
+    if first_second != 1 or whole_second != 1 or whole_first != 1:
+        return False, f"For {key}:\n is not stable\n"
+    return True, f"For {key}: is stable\n"
+
+
+def similarity_test(embeddings: dict) -> dict:
+    """
+    Compute similarity of columns based on embeddings similarity/distance. It should be 1 for same columns.
+    Result from this function will be used in generate_report for counting similar and not similar score.
+    :param embeddings: embeddings of columns
+    """
+    keys = list(embeddings.keys())
+    res = dict()
+    # Initialize nested dictionaries for each function
+    for function in embeddings[list(embeddings.keys())[0]].keys():
+        res[function] = dict()
+
+    for column in embeddings.keys():
+        for function in embeddings[column].keys():
+            for column2 in keys:
+                sim = cosine_sim(embeddings[column][function], embeddings[column2][function])
+                res[function][column, column2] = sim
+                # res[function][column2, column] = sim
+        # keys.remove(column)
+    return res
+
+
+def time_test(column1: pd.Series, function, key: str) -> (bool, float):
+    """
+    Computes how much time it takes to compute embeddings for column using specific function.
+    Cache are not used for this test
+    :param column1: column to test
+    :param function: function to use
+    :param key: key for cache
+    """
+    # cache.set_file("no_file.txt")
+    # cache.clear_cache()
```

Comment:
Commented out code either has to have a good explanation next to it or should not be there.

Reply:
Fixed